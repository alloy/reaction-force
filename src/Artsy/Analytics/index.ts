import * as Schema from "./Schema"

export * from "./track"
export * from "./Schema"
export * from "./TrackingContext"
export const AnalyticsSchema = Schema
