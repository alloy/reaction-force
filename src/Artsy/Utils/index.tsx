export { ClientRouter } from "./ClientRouter"
export { MockBoot } from "./MockBoot"
export { MockRelayRenderer } from "./MockRelayRenderer"
