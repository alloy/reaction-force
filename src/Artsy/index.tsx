export * from "./Analytics"
export { ContextProvider, ContextConsumer, ContextProps } from "./SystemContext"
