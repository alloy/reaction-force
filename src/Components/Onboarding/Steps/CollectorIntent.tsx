import React from "react"
import { commitMutation, graphql } from "react-relay"
import styled from "styled-components"

import { ContextConsumer, ContextProps } from "../../Artsy"
import { media } from "../../Helpers"
import SelectableLink from "../SelectableLink"
import { StepProps } from "../Types"
import { Layout } from "./Layout"

const OptionsContainer = styled.div`
  width: 450px;
  margin: 0 auto 100px;
  &:last-child {
    border-bottom: 1px solid #e5e5e5;
  }
  ${media.sm`
    width: 100%;
    margin-bottom: 20px;
  `};
`

type Props = StepProps & ContextProps

interface State {
  selectedOptions: { [option: string]: boolean }
  selectedCount: number
  error?: string
}

class CollectorIntent extends React.Component<Props, State> {
  options = [
    "buy art & design",
    "sell art & design",
    "research art prices",
    "learn about art",
    "find out about new exhibitions",
    "read art market news",
  ]

  intentEnum = {
    "buy art & design": "BUY_ART_AND_DESIGN",
    "sell art & design": "SELL_ART_AND_DESIGN",
    "research art prices": "RESEARCH_ART_PRICES",
    "learn about art": "LEARN_ABOUT_ART",
    "find out about new exhibitions": "FIND_ART_EXHIBITS",
    "read art market news": "READ_ART_MARKET_NEWS",
  }

  constructor(props) {
    super(props)

    this.state = {
      selectedOptions: {},
      selectedCount: 0,
    }
  }

  onOptionSelected = (index, selected) => {
    const option = this.options[index]

    let selectedOptions = this.state.selectedOptions
    selectedOptions[option] = selected

    let count = 0
    for (let key in selectedOptions) {
      if (selectedOptions[key]) count++
    }

    this.setState({ selectedOptions, selectedCount: count })
  }

  submit() {
    const keys = Object.keys(this.state.selectedOptions)
    const intentOptions = keys.filter(key => {
      return this.state.selectedOptions[key]
    })

    const intents = intentOptions.map(intent => {
      return this.intentEnum[intent]
    })

    commitMutation(this.props.relayEnvironment, {
      mutation: graphql`
        mutation CollectorIntentUpdateCollectorProfileMutation(
          $input: UpdateCollectorProfileInput!
        ) {
          updateCollectorProfile(input: $input) {
            intents
          }
        }
      `,
      variables: {
        input: {
          intents,
        },
      },
    })
  }

  render() {
    const options = this.options.map((text, index) => (
      <SelectableLink
        key={index}
        text={text}
        onSelect={this.onOptionSelected.bind(this, index)}
      />
    ))
    return (
      <Layout
        title="Get started on Artsy, what are you most interested in doing?"
        subtitle="Select all that apply"
        onNextButtonPressed={
          this.state.selectedCount > 0 && this.submit.bind(this)
        }
      >
        <OptionsContainer>{options}</OptionsContainer>
      </Layout>
    )
  }
}

export default ContextConsumer(CollectorIntent)
