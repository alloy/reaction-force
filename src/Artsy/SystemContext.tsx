import React, { SFC, useContext, useMemo, useState } from "react"
import { Environment } from "relay-runtime"

import { createRelaySSREnvironment } from "Artsy/Relay/createRelaySSREnvironment"
import { getUser } from "Utils/user"

export interface Mediator {
  trigger: (action: string, config?: object) => void
}

/**
 * Globally accessible SystemContext values for use in Artsy apps
 */
export interface SystemContextProps {
  /** Is the user opening a Reaction page from the mobile app */
  isEigen?: boolean

  /**
   * When a request is being performed by the router this value is updated
   */
  isRouterFetching?: boolean

  /**
   * A PubSub hub, which should only be used for communicating with Force.
   */
  mediator?: Mediator

  /**
   * FIXME: Ask alloy how to pass one-off props like this in from force
   */
  notificationCount?: number
  searchQuery?: string

  /**
   * A configured environment object that can be used for any Relay operations
   * that need an environment object.
   *
   * If none is provided to the `SystemContextProvider` then one is created,
   * using the `user` if available.
   */
  relayEnvironment?: Environment

  /**
   * Toggles `isRouterFetching`, which UI can tap into for displaying custom
   * loading states.
   */
  setRouterFetching?: (isFetching: boolean) => void

  /**
   * The currently signed-in user.
   *
   * Unless explicitely set to `null`, this will default to use the `USER_ID`
   * and `USER_ACCESS_TOKEN` environment variables if available.
   */
  user?: User
}

export const SystemContext = React.createContext<SystemContextProps>({})

/**
 * Creates a new Context.Provider with a user and Relay environment, or defaults
 * if not passed in as props.
 */
export const SystemContextProvider: SFC<SystemContextProps> = ({
  children,
  ...props
}) => {
  const [isRouterFetching, setRouterFetching] = useState(false)
  const user = getUser(props.user)

  const relayEnvironment =
    props.relayEnvironment || createRelaySSREnvironment({ user })

  console.log(isRouterFetching, "-------------------")
  const providerValues = useMemo(() => {
    return {
      ...props,
      isRouterFetching,
      relayEnvironment,
      setRouterFetching,
      user,
    }
  }, [props.relayEnvironment, props.user])

  return (
    <SystemContext.Provider value={providerValues}>
      {children}
    </SystemContext.Provider>
  )
}

export const SystemContextConsumer = SystemContext.Consumer

/**
 * A HOC utility function for injecting renderProps into a component.
 */
export const withSystemContext = Component => {
  return props => {
    return (
      <SystemContextConsumer>
        {contextValues => {
          return <Component {...contextValues} {...props} />
        }}
      </SystemContextConsumer>
    )
  }
}

/**
 * Custom hook to access SystemContext
 */
export const useSystemContext = () => {
  const systemContext = useContext(SystemContext)
  return systemContext
}
