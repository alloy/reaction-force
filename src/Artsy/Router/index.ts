import { HistoryOptions, HistoryProtocol } from "farce"
import { RouteConfig } from "found"
import { ContextProps } from "../SystemContext"

// API Exports for either client or server
export { PreloadLink, PreloadLinkProps } from "./Components/PreloadLink"
export { Link } from "found"
export { Boot } from "./Components/Boot"
export { ContextProvider, ContextConsumer } from "Artsy/SystemContext"

/**
 * Configuration used when creating a new Router app
 */
export interface RouterConfig {
  /**
   * Context values to be passed to ArtsyContext
   */
  context?: ContextProps

  /**
   * Configuration options to be passed to Found router
   */
  history?: {
    /**
     * Defines the type of history to use, depending on router environment.
     */
    protocol?: HistoryProtocol

    options?: HistoryOptions
  }

  /**
   * Initial route for router on boot
   */
  initialRoute?: string

  /**
   * Array of routes to be passed to Found
   */
  routes: RouteConfig[]

  /**
   * URL passed from server
   */
  url?: string
}
