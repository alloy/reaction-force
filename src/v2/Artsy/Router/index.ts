import { RelaySSREnvironment } from "v2/Artsy/Relay/createRelaySSREnvironment"
import { HistoryOptions, HistoryProtocol } from "farce"
import { RouteConfig } from "found"
import { SystemContextProps } from "../SystemContext"

export { Link } from "found"
export { Boot } from "./Boot"
export { SystemContextProvider, SystemContextConsumer } from "v2/Artsy"

/**
 * Configuration used when creating a new Router app
 */
export interface RouterConfig {
  /**
   * Context values to be passed to ArtsyContext
   */
  context?: SystemContextProps & { relayEnvironment?: RelaySSREnvironment }

  /**
   * Configuration options to be passed to Found router
   */
  history?: {
    /**
     * Defines the type of history to use, depending on router environment.
     */
    protocol?: HistoryProtocol

    options?: HistoryOptions
  }

  /**
   * Initial route for router on boot
   */
  initialRoute?: string

  /**
   * Array of routes to be passed to Found
   */
  routes: RouteConfig[]

  /**
   * URL passed from server
   */
  url?: string
}
