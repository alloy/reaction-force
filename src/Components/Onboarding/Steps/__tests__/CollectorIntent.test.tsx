import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { ContextProvider } from "../../../Artsy"
import CollectorIntent from "../CollectorIntent"

describe("CollectorIntent", () => {
  describe("snapshot", () => {
    it("renders the collector intent step", () => {
      const component = renderer
        .create(
          <ContextProvider>
            <CollectorIntent onNextButtonPressed={jest.fn()} />
          </ContextProvider>
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
