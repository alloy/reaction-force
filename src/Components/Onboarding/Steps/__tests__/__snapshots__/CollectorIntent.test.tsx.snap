// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectorIntent snapshot renders the collector intent step 1`] = `
.c8 {
  font-family: artsy-icons;
  color: white;
  font-size: 39px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c7 {
  width: 18px;
  height: 18px;
  background-color: black;
  border-radius: 50%;
  float: right;
  margin-right: 15px;
  display: none;
  -webkit-box-pack: none;
  -webkit-justify-content: none;
  -ms-flex-pack: none;
  justify-content: none;
}

.c5 {
  display: block;
  font-size: 14px;
  color: black;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 30px 0 30px 15px;
  border-top: 1px solid #e5e5e5;
}

.c5:hover {
  background-color: #f8f8f8;
  cursor: pointer;
}

.c5:hover .c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c11 {
  background: #e5e5e5;
  color: rgba(0,0,0,0.5);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c11:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c10 {
  background: white;
  color: rgba(0,0,0,0.5);
  border: 1px solid #e5e5e5;
}

.c10:hover:not(:disabled) {
  background: white;
  color: #6e1fff;
}

.c2 {
  font-size: 50px;
  color: inherit;
  margin: 20px 0;
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
}

.c0 {
  max-width: 930px;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  text-align: center;
  line-height: normal;
}

.c3 {
  color: #999999;
  margin-bottom: 100px;
  text-align: center;
  line-height: normal;
}

.c9 {
  margin: 0 auto 50px;
  display: block;
  width: 250px;
}

.c4 {
  width: 450px;
  margin: 0 auto 100px;
}

.c4:last-child {
  border-bottom: 1px solid #e5e5e5;
}

@media (max-width:48em) {
  .c5 {
    padding: 15px 0 15px 15px;
  }
}

@media (max-width:48em) {
  .c2 {
    font-size: 25px;
  }
}

@media (max-width:48em) {
  .c0 {
    margin: 20px;
  }
}

@media (max-width:48em) {
  .c1 {
    text-align: left;
  }
}

@media (max-width:48em) {
  .c3 {
    text-align: left;
    margin-bottom: 15px;
    font-size: 20px;
  }
}

@media (max-width:48em) {
  .c9 {
    width: 100%;
  }
}

@media (max-width:48em) {
  .c4 {
    width: 100%;
    margin-bottom: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    color="inherit"
  >
    Get started on Artsy, what are you most interested in doing?
     
  </div>
  <div
    className="c3 c2"
    color="inherit"
  >
    Select all that apply
  </div>
  <div>
    <div
      className="c4"
    >
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          buy art & design
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          sell art & design
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          research art prices
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          learn about art
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          find out about new exhibitions
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
      <div>
        <a
          className="c5"
          href={undefined}
          onClick={[Function]}
        >
          read art market news
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={undefined}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
  <button
    className="c9 c10 c11"
    disabled={true}
    onClick={null}
  >
    <span>
      Next
    </span>
  </button>
</div>
`;
